app/controllers/users_controller.rb

```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+
+  get '/signup' do
+    if !!session[:id]
+      redirect to "/tweets"
+    else 
+      erb :'users/create_user'
+    end
+  end
+
+  post '/signup' do
+    if params.has_value?("")
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.